File: uikit/src/main/java/com/sendbird/uikit/model/MutableLiveDataEx.java
package com.sendbird.uikit.model;

/**
 * LiveData which publicly exposes setValue(T) and postValue(T) method.
 *
 * @param <T> The type of data hold by this instance
 * since 3.0.0
 */
public class MutableLiveDataEx<T> extends LiveDataEx<T> {
    public MutableLiveDataEx(T value) {
        super(value);
    }

    public MutableLiveDataEx() {
        super();
    }

    @Override
    public void postValue(T value) {
        super.postValue(value);
    }

    @Override
    public void setValue(T value) {
        super.setValue(value);
    }
}

File: uikit/src/main/java/com/sendbird/uikit/model/MutableLiveDataEx.kt
package com.sendbird.uikit.model

/**
 * LiveData which publicly exposes setValue(T) and postValue(T) method.
 *
 * @param <T> The type of data hold by this instance
 * @since 3.0.0
 */
class MutableLiveDataEx<T> : LiveDataEx<T> {
    constructor(value: T) : super(value)
    constructor() : super()

    public override fun postValue(value: T) {
        super.postValue(value)
    }

    public override fun setValue(value: T) {
        super.setValue(value)
    }
}
